Dockerfile

```diff
@@ -194,11 +207,18 @@ RUN --mount=type=bind,from=build,src=/workspace/dist,target=/vllm-workspace/dist
     --mount=type=cache,target=/root/.cache/pip \
     python3 -m pip install dist/*.whl --verbose
 
-RUN --mount=type=cache,target=/root/.cache/pip \
-. /etc/environment && \
-if [ "$TARGETPLATFORM" != "linux/arm64" ]; then \
-    python3 -m pip install https://github.com/flashinfer-ai/flashinfer/releases/download/v0.1.6/flashinfer-0.1.6+cu121torch2.4-cp${PYTHON_VERSION_STR}-cp${PYTHON_VERSION_STR}-linux_x86_64.whl; \
-fi
+# Install FlashInfer wheel
+# NOTE: FlashInfer's wheel is not AOT compiled for 0.2.0, so we will build AOT from source in `base` stage
+RUN --mount=type=bind,from=build,src=/workspace/flashinfer/dist,target=/vllm-workspace/flashinfer-dist \
+    --mount=type=cache,target=/root/.cache/pip \
+    python3 -m pip install flashinfer-dist/*.whl --verbose
+
+# RUN --mount=type=cache,target=/root/.cache/pip \
+# . /etc/environment && \
+# if [ "$TARGETPLATFORM" != "linux/arm64" ]; then \
+#     python3 -m pip install https://github.com/flashinfer-ai/flashinfer/releases/download/v0.2.0.post1/flashinfer-0.2.0.post1+cu121torch2.4-cp${PYTHON_VERSION_STR}-cp${PYTHON_VERSION_STR}-linux_x86_64.whl; \
+# fi
```

Comment:
delete prior to landing?

Reply:
commented in [bb44221](https://github.com/vllm-project/vllm/pull/11194/commits/bb4422161e0cd310815e4c09830898258245dd42) . we might need it soon, so I just comment it out right now.